junit-platform-commons/src/main/java/module-info.java

```diff
@@ -8,15 +8,20 @@
  * https://www.eclipse.org/legal/epl-v20.html
  */
 
+import org.jspecify.annotations.NullMarked;
+
 /**
  * Common APIs and support utilities for the JUnit Platform.
  *
  * @since 1.0
  */
+@NullMarked
 module org.junit.platform.commons {
 	requires java.logging;
 	requires java.management; // needed by RuntimeUtils to determine input arguments
+
 	requires static transitive org.apiguardian.api;
+	requires static transitive org.jspecify;
```

Comment:
We should discuss whether this is the right "scope". It requires downstream projects  to have the JSpecify jar on their module path (should they be compiled using the module path).

Reply:
`requires static` would make it optional, but we'd have to suppress [warnings like these](https://ge.junit.org/s/bcfbrjy3gr5yk/console-log/task/:junit-platform-commons:compileJava?anchor=45&page=1).

Reply:
Mh, yeah. It ships with a compiled module descriptor, though: [org.jspecify](https://github.com/sormuras/modules/blob/d2d9fc919c4ab285c40ef75228887d88ff76c40a/com.github.sormuras.modules/com/github/sormuras/modules/modules.properties#L6374C1-L6374C97)